core/functionalTest.gradle

```diff
@@ -0,0 +1,21 @@
+// Stolen from
+// https://github.com/gradle-guides/testing-gradle-plugins/blob/master/samples/code/url-verifier-plugin/gradle/functional-test.gradle
+
+sourceSets {
```

Comment:
I'm also not 100% happy with that solution.
But I've read that https://guides.gradle.org/testing-gradle-plugins/ and IMO everything makes sense to me (having a test, functionalTest and integrationTest sourceSet).
But yes, we can move it...

About the `DefaultGradleRunner`:
No, the custom sourceSet is not needed to use the GradleRunner (DefaultGradleRunner is a internal API and **should not be used!**). We also apply the `gradleTestKit()`, but it is a little bit hidden ;)
Since we are using the `java-gradle-plugin` it will be automatically applied for us.
See https://docs.gradle.org/3.3/userguide/javaGradle_plugin.html:
> It automatically adds the gradleTestKit() dependency to the test compile configuration and generates a plugin classpath manifest file consumed by a GradleRunner instance if found.

But we have "another" issue with the PR in general.
I will explain it later in a upcoming commit.. 👍

Reply:
I see. Thanks for the clarification. I just was curious why we need this setup.